django/db/backends/base/schema.py

```diff
@@ -679,6 +685,13 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
             fragment, other_actions = self._alter_column_type_sql(model, old_field, new_field, new_type)
             actions.append(fragment)
             post_actions.extend(other_actions)
+        # Collation change?
+        if hasattr(old_field, 'db_collation') or hasattr(new_field, 'db_collation'):
+            old_collation = getattr(old_field, 'db_collation', None)
+            new_collation = getattr(new_field, 'db_collation', None)
+            if old_collation != new_collation:
+                fragment = self._alter_column_collation_sql(new_field, new_type, new_collation)
+                actions.append(fragment)
```

Comment:
Can this be simplified to

```suggestion
        # Collation change?
        elif getattr(old_field, 'db_collation', None) != getattr(new_field, 'db_collation', None):
            new_collation = getattr(new_field, 'db_collation', None)
            fragment = self._alter_column_collation_sql(new_field, new_type, new_collation)
            actions.append(fragment)
```

Or maybe we want to keep the old code to support field type and collation change at the same time (e.g. `CharField(db_collation='foo') -> TextField(db_collation='bar')`).

Reply:
Yeah, that's the case I was thinking about. I think an extra query is okay here. I don't see it as very likely that someone will change the type and collation at the same time, but if they do, it'd be very confusing that it doesn't work, I think. I'm just a little paranoid of weirdness there.